README
This extention contains some CnC TA scripts writen by many authors. I take no credit on any script in this extension!

====================
This extension contains scripts listed bellow:

- CnC:Tiberium Alliances Shortcut 1.8.3
- C&C:TA CNCOpt Link Button 1.7.6
- Tiberium Alliances Formation Saver 2.1.8
- CnC: MH Tiberium Alliances Available Loot Summary 1.8.3
- MaelstromTools Dev 0.1.3.2
- CnC: MH Tiberium Alliances Pure Loot Summary 1.7.2p
- TACS (Tiberium Alliances Combat Simulator) 2.2 & 3.0b
- infernal wrapper 0.390737.5
- Tiberium Alliances Combat Simulator (Basic) 0.3.1
- C&C Combat Simulator 0.4
- C&C:Tiberium Alliances Maelstrom ADDON Basescanner 1.8.4
- Tiberium Alliances Map and Zoom 1.1
- CnC Tiberium Coord Box Shortcut 1.1
- CnC: Tiberium Alliances Map (KSX-Mod) 1.5
- C&C:Tiberium Alliances Extended Chathelper 1.0.11
- C&C:Tiberium Alliances Coords Button ALL 2.0.1
- C&C: TA Compass Movable 1.1.0
- C&C: Tiberium Alliances Chat Helper Enhanced 3.0.0
- CustomFlunik Tools 20130415a
- The Green Cross - Tiberium Alliances Combat Simulator 3.2.2
- Tiberium Alliances PvP/PvE Player Info Mod 1.2
- The Green Cross - Tiberium Alliances Tools 0.5
- Tiberium Alliances Transfer All Resources 1.6.1
- CnC: Tiberium Alliances Info 1.0.3
- Tiberium Alliances Info Sticker 1.11.1
- CnC:Tiberium Aliances Navigator - Compass 1.32
- C&C Tiberium Alliances Flunik Tools: Custom AutoUpgrade 0.5.7.09
- CnC: Tiberium Alliances Available Loot Summary + Info 2.0.0
- Tiberium Alliances BaseNavBar Reorderer 1.0
- WarChiefs - Tiberium Alliances Combat Simulator 13.07.03
- Tiberium Alliances - New Resource Trade Window 1.4.7
- C&C Tiberium Alliances Flunik Tools: New Custom AutoUpgrade 'Awesome 1.2.5'
- CnC: Tiberium Alliances Shortcuts 1.8.1
- TA - Upgrade Tool 1.0
- Command & Conquer TA World Map 1.0.0
- Command and Conquer TA POIs Analyser 1.5.3
- WarChiefs - Tiberium Alliances Sector HUD 13.07.05
- WarChiefs - Tiberium Alliances Upgrade Base/Defense/Army 13.07.20


CHANGELOG
1.2.8.59:
- Update "TACS" to version 3.07b
1.2.8.58:
 - Various updates based on cncscripts.com
1.2.8.56:
- Minor bug fix
1.2.8.55:
 - Add "xTr1m's Base Overlay" version 1.0
1.2.8.54:
- Add "C&C:TA Count Forgotten Bases in Range" version 1.1.0
1.2.8.53:
- Update Google Analytics tracking code
1.2.8.52:
- Add back "C&C:Tiberium Alliances Maelstrom ADDON Basescanner"
- Add back "MaelstromTools Dev"
1.2.8.51:
- Use UserScript mirror link
- Remove TACS 2.2
- Remove "C&C:Tiberium Alliances Maelstrom ADDON Basescanner"
- Remove "MaelstromTools Dev"
- Remove "Tiberium Alliances Combat Simulator (Basic)"
1.2.8.50:
- Alot of scripts are up-to-date
1.2.8.49:
- Fix: CPCOpt not working
1.2.8.48:
- Update "C&C:TA CNCOpt Link Button" to version 1.7.6 (support new worlds, by anoek)
1.2.8.47:
- Add "WarChiefs - Tiberium Alliances Sector HUD" version 13.07.05
- Add "WarChiefs - Tiberium Alliances Upgrade Base/Defense/Army" version 13.07.20
- Update "Command & Conquer TA POIs Analyser" to version 1.5.3
- Update "Tiberium Alliances PvP/PvE Player Info Mod" version 1.2
- Update "C&C Tiberium Alliances Flunik Tools: New Custom AutoUpgrade" to version Awesome 1.2.5
1.2.8.46:
- Remove "C&C:Tiberium Alliances Maelstrom ADDON Citycolor"
- Update "CnC: Tiberium Alliances Shortcuts" to version 1.8.1
- Update "CnC Tiberium Coord Box Shortcut" to version 1.1
- Update "WarChiefs - Tiberium Alliances Combat Simulator" version 13.07.03
- Update "C&C Tiberium Alliances Flunik Tools: New Custom AutoUpgrade" to version Awesome 1.2.4
- Add "TA - Upgrade Tool" version 1.0
- Add "Command & Conquer TA World Map" version 1.0.0
- Add "Command and Conquer TA POIs Analyser" version 1.5.2
1.2.8.45:
- Update "Tiberium Alliances - New Resource Trade Window" to version 1.4.7
1.2.8.44:
- Update "Tiberium Alliances Transfer All Resources" to version 1.6.1
1.2.8.43:
- Add "C&C Tiberium Alliances Flunik Tools: New Custom AutoUpgrade" version Awesome 1.2.2
- Update "CnC:Tiberium Aliances Navigator - Compass" to version 1.32
1.2.8.42:
- Add "Tiberium Alliances - New Resource Trade Window" version 1.2
- Update "C&C:Tiberium Alliances Coords Button" to version 2.0.1 (Updated by Chiantii)
1.2.8.41:
- Update "C&C:Tiberium Alliances Coords Button" to version 2.0 (Updated by Chiantii)
1.2.8.40:
- Add "nWo - Tiberium Alliances Combat Simulator" version 13.05.12
1.2.8.39:
- Update "C&C:Tiberium Alliances Maelstrom ADDON Basescanner" to version 1.8.4
1.2.8.38:
- Add "Tiberium Alliances BaseNavBar Reorderer" version 1.0
1.2.8.37:
- Add "CnC: Tiberium Alliances Available Loot Summary + Info" version 2.0.0 (disabled by default)
1.2.8.36:
- Small hack to make TACS 3.0 works w/o GM
- Update "C&C Tiberium Alliances Flunik Tools: Custom AutoUpgrade" to version 0.5.7.09
- Update "CustomFlunik Tools" to version 20130415a
1.2.8.35:
- Remove duplicate TACS 2
1.2.8.34:
- Add "TACS 3 (Tiberium Alliances Combat Simulator)" 3.0b (disabled by default)
1.2.8.33:
- Update "MaelstromTools Dev" to version 0.1.3.2
- Update "C&C Tiberium Alliances Flunik Tools: Custom AutoUpgrade" to version 0.5.7.08
1.2.8.32:
- Update "CnC:Tiberium Aliances Navigator - Compass" to version 1.2.1
- Update "C&C Tiberium Alliances Flunik Tools: Custom AutoUpgrade" to version 0.5.7.05
- Update "CustomFlunik Tools" to version 20130325d
1.2.8.31:
- Update "C&C Tiberium Alliances Flunik Tools: Custom AutoUpgrade" to version 0.5.7.03
- Update "C&C:Tiberium Alliances Maelstrom ADDON Basescanner" to version 1.8.3
- Update "CustomFlunik Tools" to version 20130318c
1.2.8.30:
- Update "CustomFlunik Tools" to version 20130307c
1.2.8.29:
- Update "TACS (Tiberium Alliances Combat Simulator)" to version 2.2
1.2.8.28:
- Update "The Green Cross - Tiberium Alliances Combat Simulator" to version 3.2.2
- Update "Tiberium Alliances Info Sticker" to version 1.11.1
- Add "C&C Tiberium Alliances Flunik Tools: Custom AutoUpgrade" version 0.5.7.02
1.2.8.27:
- Add "CnC:Tiberium Aliances Navigator - Compass" version 1.1.0 (recommended, disables by default)
- Update "CustomFlunik Tools" to version 20130305a
1.2.8.26:
- Update "Tiberium Alliances Info Sticker" to version 1.10
- Remove tracking code from "Maelstrom ADDON Citycolor"
- Update "C&C:TA CNCOpt Link Button" to version 1.7.5 (support new worlds, by anoek)
- Re-add "C&C:Tiberium Alliances Maelstrom ADDON Basescanner" version 1.7.1
1.2.8.25:
- Remove "C&C:Tiberium Alliances Maelstrom ADDON Basescanner"
- Update "Tiberium Alliances Info Sticker" to version 1.9
- Update "CustomFlunik Tools" to version 20130301d
1.2.8.24:
- Update "Tiberium Alliances Info Sticker" to version 1.7.1
1.2.8.23:
- Fix Map KSX-mod (suggested by barius https://github.com/barius)
- Add Spanish translation for  "MaelstromTools Dev" (by rubenlagus https://github.com/rubenlagus)
- Replace "Flunik Tools" with "CustomFlunik Tools" version 20130301a (by towster https://github.com/towster)
1.2.8.22:
- Update "MaelstromTools Dev" to version 0.1.3.1
- Update "Tiberium Alliances Transfer All Resources" to version 1.5.1
- Add "CnC: Tiberium Alliances Info" version 1.0.3
- Add "Tiberium Alliances Info Sticker repair time" version 1.13
1.2.8.21:
- Replace "Flunik Tools" with Infernal's version 0.5.8.infernal
- Update "infernal wrapper" to version 0.390737.5
- Update "MaelstromTools Dev" to version 0.1.3.0
- Update "The Green Cross - Tiberium Alliances Tools" 0.5
1.2.8.20:
- Update "Flunik Tools" to version 0.5.7.01 (patched by dbendure)
- Update "infernal wrapper" to version 0.390737.2
- Update "MaelstromTools Dev" to version 0.1.2.8 pre
- Add "Tiberium Alliances PvP/PvE Player Info Mod" version 1.1
- Add "The Green Cross - Tiberium Alliances Tools" 0.4
1.2.8.19:
- Fix "TACS (Tiberium Alliances Combat Simulator)" script
1.2.8.18:
- Add "Tiberium Alliances Transfer All Resources" version 1.2
- Update "infernal wrapper" to version 0.389314
- Update "MaelstromTools Dev" to version 0.1.2.5 pre
- Update "C&C:Tiberium Alliances Maelstrom ADDON Basescanner" version 1.7.1
1.2.8.17:

src/js/background.js
var DEFAULT_SCRIPTS = [{
    id : 131289,
    name : "C&C:TA CNCOpt Link Button",
    version : "1.7.7",
    enabled : true
}, {
    id : 136299,
    name : "Tiberium Alliances Formation Saver",
    version : "2.2",
    enabled : false
}, {
    id : 138212,
    name : "TACS 3 (Tiberium Alliances Combat Simulator)",
    version : "3.07b",
    enabled : true
}, {
    id : 137978,
    name : "CnC: MH Tiberium Alliances Available Loot Summary",
    version : "1.8.3",
    enabled : true
}, {
    id : 145657,
    name : "CnC: MH Tiberium Alliances Pure Loot Summary",
    version : "1.7.2p",
    enabled : false
}, {
    id : 140991,
    name : "MaelstromTools Dev",
    version : "0.1.4.4",
    enabled : true
}, {
    id : 140988,
    name : "infernal wrapper",
    version : "0.390737.6",
    enabled : true
}, {
    id : 147335,
    name : "C&C Combat Simulator",
    version : "0.4",
    enabled : false
}, {
    id : 145168,
    name : "C&C:Tiberium Alliances Maelstrom ADDON Basescanner",
    version : "1.8.5",
    enabled : true
}, {
    id : 138386,
    name : "CnC Tiberium Coord Box Shortcut",
    version : "1.1",
    enabled : false
}, {
    id: 149093,
    name: "CnC: Tiberium Alliances Map (KSX-Mod)",
    version: "1.5",
    enabled: true
}, {
    id: 151965,
    name: "C&C:Tiberium Alliances Extended Chathelper",
    version: "1.0.11",
    enabled: true
}, {
    id: 153193,
    name: "C&C:TA Compass Movable",
    version: "1.1.0",
    enabled: true
}, {
    id: 152177,
    name: "C&C:Tiberium Alliances Extended Chathelper Enhanced",
    version: "3.1.6",
    enabled: false
}, {
    id: 158565,
    name: "CustomFlunik Tools",
    version: "20130415a",
    enabled: true
}, {
    id: 154546,
    name: "The Green Cross - Tiberium Alliances Combat Simulator",
    version: "3.2.2",
    enabled: false
}, {
    id: 158164,
    name: "Tiberium Alliances Transfer All Resources",
    version: "1.6.1",
    enabled: true
}, {
    id: 158919,
    name: "The Green Cross - Tiberium Alliances Tools",
    version: "0.5",
    enabled: true
}, {
    id: 159150,
    name: "Tiberium Alliances PvP/PvE Player Info Mod",
    version: "1.3",
    enabled: true
}, {
    id: 149251,
    name: "CnC: Tiberium Alliances Info",
    version: "1.0.3",
    enabled: false
}, {
    id: 155157,
    name: "Tiberium Alliances Info Sticker",
    version: "1.11.1",
    enabled: false
}, {
    id: 158869,
    name: "C&C Tiberium Alliances Flunik Tools",
    version:  "Awesome 1.2.6",
    enabled: false
}, {
    id : 135806,
    name : "PluginsLib - mhShortcuts - Tiberium Alliances",
    version : "1.8.3",
    enabled : false
}, {
    id: 159496,
    name: "PluginsLib - mhNavigator - Tiberium Alliances",
    version: "1.35",
    enabled: false
}, {
    id : 160800,
    name : "PluginsLib - mhLoot - Tiberium Alliances",
    version : "2.06",
    enabled : false
}, {
    id : 165888,
    name : "WarChiefs - Tiberium Alliances Combat Simulator",
    version : "13.09.26",
    enabled : false
}, {
    id : 168297,
    name : "Tiberium Alliances - New Resource Trade Window",
    version : "1.4.9",
    enabled : false
}, {
    id : 166833,
    name : "New Custom Flunik Tools",
    version : "Awesome 1.3.3",
    enabled : false
}, {
    id : 158799,
    name : "TA - Upgrade Tool",
    version : "1.0",
    enabled : false
}, {
    id : 173330,
    name : "Command & Conquer TA World Map",
    version : "1.0.0",
    enabled : false
}, {
    id : 171353,
    name : "Command and Conquer TA POIs Analyser",
    version : "2.0.1",
    enabled : false
}, {
    id : 167564,
    name : "WarChiefs - Tiberium Alliances Upgrade Base/Defense/Army",
    version : "13.10.30",
    enabled : false
}, {
    id : 172683,
    name : "WarChiefs - Tiberium Alliances Sector HUD",
    version : "13.12.18",
    enabled : false
}, {
    id : 3343,
    name : "C&C:TA Count Forgotten Bases in Range",
    link : "https://greasyfork.org/en/scripts/3343-c-c-ta-count-forgotten-bases-in-range",
    version : "1.1.0",
    enabled : false
}, {
    id : 10382,
    name : "xTr1m's Base Overlay",
    link : "https://greasyfork.org/en/scripts/10382-c-c-tiberium-alliances-xtr1m-s-base-overlay",
    version : "1.0",
    enabled : false
}];

var storage = chrome.storage.sync;

storage.get(['CNCTA_VERSION', 'CNCTA_ENABLED', 'CNCTA_GA'], function (config) {
    if (chrome.app.getDetails().version !== config.CNCTA_VERSION) {
        window.open(chrome.extension.getURL('updated.html'));
        if (!config.CNCTA_ENABLED) {
            config.CNCTA_ENABLED = {};
        }
        var tmp = {};

        for (var i in DEFAULT_SCRIPTS) {
            if (DEFAULT_SCRIPTS.hasOwnProperty(i)) {
                var script = DEFAULT_SCRIPTS[i];
                if (typeof config.CNCTA_ENABLED['s_' + script.id] === 'undefined' || config.CNCTA_ENABLED['s_' + script.id] === null) {
                    tmp['s_' + script.id] = script.enabled;
                } else {
                    tmp['s_' + script.id] = config.CNCTA_ENABLED['s_' + script.id];
                }
            }
        }
        storage.set({
            'CNCTA_VERSION': chrome.app.getDetails().version,
            'CNCTA_SCRIPTS': DEFAULT_SCRIPTS,
            'CNCTA_ENABLED': tmp
        });
    }

    if (!config.CNCTA_GA) {
        storage.set({'CNCTA_GA': true});
        config.CNCTA_GA = true;
    }
    if (config.CNCTA_GA === true) {
        (function () {
            var ga = document.createElement('script');
            ga.type = 'text/javascript';
            ga.async = true;
            ga.src = 'https://ssl.google-analytics.com/analytics.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(ga, s);
        })();
        window.ga = window.ga || function () {
            (ga.q = ga.q || []).push(arguments);
        };
        ga.l = + new Date();
        ga('create', 'UA-15252221-7', 'auto');
        ga('set', 'checkProtocolTask', function () {});
        ga('require', 'displayfeatures');
        ga('send', 'pageview', '/');
    }
});

function processRequest(request, sender, sendResponse) {
    switch (request.type) {
    case "get":
        var data = {};
        if (request.name.constructor === Array) {
            for (var i = 0; i < request.name.length; i++) {
                data[request.name[i]] = localStorage[request.name[i]];
            }
            sendResponse(data);
        } else {
            sendResponse(localStorage[request.name]);
        }
        break;
    case "set":
        localStorage[request.name] = request.value;
        break;
    case "pageAction":
        chrome.pageAction.show(sender.tab.id);
        sendResponse({});
        break;

    default:
        console.log("invalid request " + request);
        break;
    }
}

if (typeof chrome.extension.sendMessage === 'undefined') {
    chrome.extension.onRequest.addListener(processRequest);
} else {
    chrome.extension.onMessage.addListener(processRequest);
}

/*------------------------------------------------*/
chrome.pageAction.onClicked.addListener(function () {
    window.open(chrome.extension.getURL('options.html'));
});
/*------------------------------------------------*/
